.github/workflows/when-pr-mergeable.yaml

```diff
@@ -0,0 +1,35 @@
+name: Write Comment on PR Update
+
+on:
+    pull_request:
+        types: [opened, edited, synchronize]
+
+jobs:
+    write-single-commit-commment:
+        # Only run when the pull request is mergeable and has one commit
+        if: github.event.pull_request.mergeable == true && github.event.pull_request.comments == 1
```

Comment:
Do *all* single commit changes have a single comment? What happens if I comment on a PR, wouldn't that mean there would be >1 comment?

How about "commits"? See https://docs.github.com/en/webhooks/webhook-events-and-payloads#pull_request